pkg/nsx/services/ippool/ippool.go

```diff
@@ -0,0 +1,256 @@
+package ippool
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha2"
+	commonctl "github.com/vmware-tanzu/nsx-operator/pkg/controllers/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/util"
+)
+
+var (
+	log                           = logger.Log
+	MarkedForDelete               = true
+	EnforceRevisionCheckParam     = false
+	ResourceTypeIPPool            = common.ResourceTypeIPPool
+	ResourceTypeIPPoolBlockSubnet = common.ResourceTypeIPPoolBlockSubnet
+	NewConverter                  = common.NewConverter
+)
+
+type IPPoolService struct {
+	common.Service
+	ipPoolStore            *IPPoolStore
+	ipPoolBlockSubnetStore *IPPoolBlockSubnetStore
+}
+
+func InitializeIPPool(service common.Service) (*IPPoolService, error) {
+	wg := sync.WaitGroup{}
+	wgDone := make(chan bool)
+	fatalErrors := make(chan error)
+
+	wg.Add(2)
+
+	ipPoolService := &IPPoolService{Service: service}
+	ipPoolService.ipPoolStore = &IPPoolStore{ResourceStore: common.ResourceStore{
+		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{common.TagScopeIPPoolCRUID: indexFunc}),
+		BindingType: model.IpAddressPoolBindingType(),
+	}}
+	ipPoolService.ipPoolBlockSubnetStore = &IPPoolBlockSubnetStore{ResourceStore: common.ResourceStore{
+		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{common.TagScopeIPPoolCRUID: indexFunc}),
+		BindingType: model.IpAddressPoolBlockSubnetBindingType(),
+	}}
+
+	go ipPoolService.InitializeResourceStore(&wg, fatalErrors, ResourceTypeIPPool, nil, ipPoolService.ipPoolStore)
```

Comment:
For VPCSubnet, it also has IP Pool, so this code will also initialize IPAddressPool for VPCSubnet in ippoolStore. Need to filter with tags.

Reply:
fixed.